clang/runtime/dpct-rt/include/dpct/graph.hpp

```diff
@@ -133,5 +133,75 @@ static void add_dependencies(dpct::experimental::command_graph_ptr graph,
   }
 }
 
+///  Auxiliary function to query graph nodes.
+/// \param [in] graph A pointer to the command graph.
+/// \param [in] get_func Lambda function for overloading
+/// \param [in] expectednumberOfNodes The number of nodes
+/// to be added
+inline std::vector<sycl::ext::oneapi::experimental::node> get_nodes_impl(
```

Comment:
It seems to be an internal function; how about putting it into `namespace detail`?

Reply:
Yes can be added there, Will update.